src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -173,11 +173,12 @@ void Terminal::UpdateSettings(winrt::Microsoft::Terminal::Settings::ICoreSetting
     {
         return S_FALSE;
     }
-    const auto dx = viewportSize.X - oldDimensions.X;
+
+    const auto dx = ::base::saturated_cast<short>(viewportSize.X - oldDimensions.X);

```

Comment:
is casting after the fact acceptable or should we be using ClampedAdd/ClampedSub?

Reply:
@miniksa I actually don't know the answer to this. The way I'm doing it seems fine, right? It'll auto promote the result to an `int`, then cast down to a `short`?

Reply:
From chat, `::base::ClampedSub(viewportSize.X, oldDimensions.X)` is the way to do this correctly